addon/webextension/background/startBackground.js

```diff
@@ -0,0 +1,94 @@
+/* globals browser, main, communication */
+/* This file handles browser.browser.onClicked, and loads the rest of the background
+   page in response to that event. */
+
+this.startBackground = (function() {
+  const backgroundScripts = [
+    "log.js",
+    "makeUuid.js",
+    "catcher.js",
+    "background/selectorLoader.js",
+    "background/communication.js",
+    "background/auth.js",
+    "background/senderror.js",
+    "build/raven.js",
+    "build/shot.js",
+    "background/analytics.js",
+    "background/deviceInfo.js",
+    "background/takeshot.js",
+    "background/main.js"
+  ];
+
+  browser.browserAction.onClicked.addListener((tab) => {
+    loadIfNecessary().then(() => {
+      main.onClicked(tab);
+    }).catch((error) => {
+      console.error("Error loading Screenshots:", error);
+    });
+  });
+
+
+  browser.contextMenus.create({
+    id: "create-screenshot",
+    title: browser.i18n.getMessage("contextMenuLabel"),
+    contexts: ["page"],
+    documentUrlPatterns: ["<all_urls>"]
+  });
+
+  browser.contextMenus.onClicked.addListener((info, tab) => {
+    loadIfNecessary().then(() => {
+      main.onClickedContextMenu(info, tab);
+    }).catch((error) => {
+      console.error("Error loading Screenshots:", error);
+    });
+  });
+
+  // Note this duplicates functionality in main.js, but we need to change
+  // the onboarding icon before main.js loads up
```

Comment:
Instead of copying the code around, what if main.js exposed a callback (um...`onOnboarding`? idk) that it called when the onboarding view was shown + dismissed? Would be nice to avoid copying code around